packages/app/src/app/overmind/effects/vscode/index.ts

```diff
@@ -468,14 +469,31 @@ export class VSCodeEffect {
 
     setFs(this.sandboxFsSync.create(sandbox));
 
-    if (isFirstLoad) {
-      this.sandboxFsSync.sync(() => {});
-    } else {
-      this.editorApi.extensionService.stopExtensionHost();
-      this.sandboxFsSync.sync(() => {
-        this.editorApi.extensionService.startExtensionHost();
-      });
+    return new Promise(resolve => {
+      if (isFirstLoad) {
+        // Do no start extension host
+        this.sandboxFsSync.sync(resolve);
+      } else {
+        this.stopExtensionHost();
+        this.sandboxFsSync.sync(resolve);
+      }
+    });
+  }
+
+  public startExtensionHost() {
+    if (this.isRunningExtensionHost) {
+      return;
+    }
+    this.editorApi.extensionService.startExtensionHost();
+    this.isRunningExtensionHost = true;
```

Comment:
We should probably find a better way to see if it's running in the future (because people can have their extension host crash etc.). But this is okay for now, maybe we should add a comment/todo to keep it in mind in the future.

Reply:
Added a comment about this at property declaration at the top of the class